DigitalOffice/Tests/TimeManagementServiceTests/Commands/CreateLeaveTimeCommandTests.cs

```diff
@@ -0,0 +1,100 @@
+using FluentValidation;
+using Moq;
+using NUnit.Framework;
+using System;
+using LT.DigitalOffice.TimeManagementService.Commands;
+using LT.DigitalOffice.TimeManagementService.Commands.Interfaces;
+using LT.DigitalOffice.TimeManagementService.Database.Entities;
+using LT.DigitalOffice.TimeManagementService.Mappers.Interfaces;
+using LT.DigitalOffice.TimeManagementService.Models;
+using LT.DigitalOffice.TimeManagementService.Repositories.Interfaces;
+using LT.DigitalOffice.TimeManagementService.Enums;
+
+namespace LT.DigitalOffice.TimeManagementServiceUnitTests.Commands
+{
+    public class CreateWorkTimeCommandTests
+    {
+        private Mock<IValidator<CreateLeaveTimeRequest>> validatorMock;
+        private Mock<IMapper<CreateLeaveTimeRequest, DbLeaveTime>> mapperMock;
+        private Mock<ILeaveTimeRepository> repositoryMock;
+
+        private ICreateLeaveTimeCommand command;
+        private CreateLeaveTimeRequest request;
+        private DbLeaveTime createdLeaveTime;
+
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            validatorMock = new Mock<IValidator<CreateLeaveTimeRequest>>();
+            mapperMock = new Mock<IMapper<CreateLeaveTimeRequest, DbLeaveTime>>();
+            repositoryMock = new Mock<ILeaveTimeRepository>();
+
+            command = new CreateLeaveTimeCommand(validatorMock.Object, mapperMock.Object, repositoryMock.Object);
+
+            request = new CreateLeaveTimeRequest()
+            {
+                LeaveType = LeaveType.SickLeave,
+                Comment = "I have a sore throat",
+                StartTime = new DateTime(2020, 7, 24),
+                EndTime = new DateTime(2020, 7, 27),
+                WorkerUserId = Guid.NewGuid()
+            };
+
+            createdLeaveTime = new DbLeaveTime()
+            {
+                Id = Guid.NewGuid(),
+                LeaveType = request.LeaveType,
+                Comment = request.Comment,
+                StartTime = request.StartTime,
+                EndTime = request.EndTime,
+                WorkerUserId = request.WorkerUserId
+            };
+        }
+
+        [Test]
+        public void FailCreateNewWorkTimeAccordingToValidator()
+        {
+            validatorMock
+                .Setup(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(false);
+
+            Assert.Throws<ValidationException>(() => command.Execute(request));
```

Comment:
It would be cool to also check that the repositoryMock method did not get called

Reply:
Fixed.